### PR TITLE
Spam_twirling

### PR DESCRIPTION
**Pull request by kidiki**
Adding a page about SPAM Twirling

### CODE REVIEW

File: docs/error-mitigation/spam-twirling.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# SPAM Twirling 
+
+State Preparation and Measurement (SPAM) Twirling is a simple error mitigation technique used to remove any preferred direction due to readout noise.  It works by diagonalizing the noise associated with measurement by randomly flipping qubits via X gates right before the measurement and flipping the corresponding measured bit classically.
```
Comment (Q-lds) [inline docs/error-mitigation/spam-twirling.md:3]: "any preferred direction" ---> "any bias"? What is a direction?

File: docs/error-mitigation/spam-twirling.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+State Preparation and Measurement (SPAM) Twirling is a simple error mitigation technique used to remove any preferred direction due to readout noise.  It works by diagonalizing the noise associated with measurement by randomly flipping qubits via X gates right before the measurement and flipping the corresponding measured bit classically. 
+A calibration factor from the diagonal noise channel is obtained by measuring quantum circuits initialized in the zero states. As a result, the expectation values of the circuit in the presence of readout noise is much closer to the noise-free one.
+
+!!! info "For the curious"
+    The technique has been developed theoretically by E. v. d. Berg et al. [1] and by A. W. R. Smith, et al. [2] and implemented within Qiskit as one of their “resilience levels” under the name “Twirled Readout Error eXtinction (TREX)”. Experimentally has been applied by Layden et al. [3] to ensure that the  symmetry requirement $Q(s'| s) = Q(s |s')$ is met. Note that this requirement is specific to the Monte Carlo algorithm and therefore irrelevant to QAOA. 
+
+
+### The algorithm
+I. Obtain calibration data (ideally over the whole device) under BFA (subroutine)
```
Comment (Q-lds) [inline docs/error-mitigation/spam-twirling.md:11]: What is BFA?

File: docs/error-mitigation/spam-twirling.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+1. Calculate calibration factors 
+2. Prepare the desired quantum circuit 
+3. Measure under BFA (subroutine)
+4. Correct the expectation values by term 
+5. Combine all corrected terms into the final expectation value
+
+#### Subroutine: Bit Flip Averaging (BFA) technique
+
+1. Divide the total number of shots into n batches and for each batch choose a set of qubits to be flipped. Note that since we usually don’t have a very accurate model of the noise and to keep it general, we choose the schedule completely at random.
+2. The bit flip before the measurement can be done by applying an X gate, absorbing the X gate into the last layer of RX rotations or propagating it to the front (and then absorbing it into the initial |+> state).
```
Comment (Q-lds) [inline docs/error-mitigation/spam-twirling.md:28]: "into the initial |+> state", didn't we start with a |0> state? Where is the |+> coming from?

File: docs/error-mitigation/spam-twirling.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+2. Prepare the desired quantum circuit 
+3. Measure under BFA (subroutine)
+4. Correct the expectation values by term 
+5. Combine all corrected terms into the final expectation value
+
+#### Subroutine: Bit Flip Averaging (BFA) technique
+
+1. Divide the total number of shots into n batches and for each batch choose a set of qubits to be flipped. Note that since we usually don’t have a very accurate model of the noise and to keep it general, we choose the schedule completely at random.
+2. The bit flip before the measurement can be done by applying an X gate, absorbing the X gate into the last layer of RX rotations or propagating it to the front (and then absorbing it into the initial |+> state). 
+3. The bit flip after the measurement is performed by applying a classical not to the outcome bitstring, which is implemented as changing the keys in the count dictionary.
```
Comment (Q-lds) [inline docs/error-mitigation/spam-twirling.md:29]: "which is implemented as changing the keys" --> "which is implemented by flipping/negating the bitstrings"? Is this correct?

File: docs/error-mitigation/spam-twirling.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+3. Measure under BFA (subroutine)
+4. Correct the expectation values by term 
+5. Combine all corrected terms into the final expectation value
+
+#### Subroutine: Bit Flip Averaging (BFA) technique
+
+1. Divide the total number of shots into n batches and for each batch choose a set of qubits to be flipped. Note that since we usually don’t have a very accurate model of the noise and to keep it general, we choose the schedule completely at random.
+2. The bit flip before the measurement can be done by applying an X gate, absorbing the X gate into the last layer of RX rotations or propagating it to the front (and then absorbing it into the initial |+> state). 
+3. The bit flip after the measurement is performed by applying a classical not to the outcome bitstring, which is implemented as changing the keys in the count dictionary. 
+4. Combine the negated counts from all batches into a single count dictionary which is then used for calculating the calibration factors or estimating the expectation values.
```
Comment (Q-lds) [inline docs/error-mitigation/spam-twirling.md:30]: "Combine the negated counts" -- > "Combine the statistics of all the negated/flipped bitstrings from all batches into a single dictionary ..." , is this the correct interpretation?

File: mkdocs.yml [hunk 6]
```diff
@@ -138,6 +138,8 @@ nav:
           - devices/rigetti-qcs.md
   - Qubit Routing:
       - qubit-routing/qaoa-qubit-routing.md
+  - Error Mitigation:
+    - error-mitigation/spam-twirling.md
```
Comment (Q-lds) [inline mkdocs.yml:142]: Did you try to build the website? The indent seems off here!

Comment (kidiki) [reply mkdocs.yml:142]: yes, it works fine for me

File: docs/error-mitigation/spam-twirling.md (hunk 2 above)
Comment (kidiki) [reply docs/error-mitigation/spam-twirling.md]: subroutine explained below

File: docs/error-mitigation/spam-twirling.md (hunk 3 above)
Comment (kidiki) [reply docs/error-mitigation/spam-twirling.md:28]: The Run circuit can be for any state, and we usually start with the layer of hadamards. But I guess it's not clear, I'll edit. thanks

File: mkdocs.yml (hunk 6 above)
Comment (kidiki) [reply mkdocs.yml]: Screenshot 2023-04-28 at 15 51 52

Comment (kidiki) [reply mkdocs.yml]: yes, it seems to be running fine but I also fixed the yml file

**Pull request merged**